docs/codelabs/github_actions.md

```diff
@@ -0,0 +1,177 @@
+- How to setup [GitHub Actions](https://github.com/features/actions)
+- How to use Please in a GitHub Actions build
+- How to use the [setup-please](https://github.com/sagikazarmark/setup-please-action) action for better integration
+
+### What if I get stuck?
+If you get stuck with GitHub Actions, check out the [official documentation](https://docs.github.com/en/free-pro-team@latest/actions).
+
+You can find usage examples of the [setup-please](https://github.com/sagikazarmark/setup-please-action) action in [this](https://github.com/sagikazarmark/todobackend-go-kit/blob/20292fc09e25196e751e087da7c5e659cd6c452f/.github/workflows/ci.yaml) repository.
+
+If you really get
+stuck you can find us on [gitter](https://gitter.im/please-build/Lobby)!
+
+## GitHub Actions
+Duration: 5
+
+GitHub Actions is an extensible CI/CD platform provided by GitHub.
+Compared to other CI/CD solutions, GitHub Actions allows you to build all kinds automations (called workflows) triggered by various events (eg. pushing code to a branch).
+
+### Setting up GitHub Actions
+
+Workflow definitions are simple YAML files stored in the `.github/workflows` directory of your repository.
+
+The following snippet triggers a workflow named `CI` whenever commits are pushed to the `master` branch:
+
+```yaml
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Test
+        run: echo "Tests passed"
+```
+
+Go ahead and add the above snippet to `.github/workflows/ci.yaml` in your project. Then go to `https://github.com/YOU/YOUR-PROJECT/actions` and observe the workflow.
+
+## Please build
+Duration: 4
+
+Now we have a project setup with GitHub Actions, it's time to start building with Please! Let's change `ci.yaml` a little:
+
+```yaml
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+
+    steps:
+      # Setup your language of choice here:
+      # https://github.com/actions/?q=setup-&type=&language=
+
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      # Run please build
+      - name: Test
+        run: ./pleasew build //...
+```
+
+Compared to the example earlier, this workflow uses the `pleasew` script to download Please and build the project.
+
+Notice the `//...` bit at the end of the command: it's necessary on GitHub Actions.
+Check [this](https://github.com/thought-machine/please/issues/1174) issue for more details.
+
+## `setup-please` action
+Duration: 10
+
+The [setup-please](https://github.com/sagikazarmark/setup-please-action) action provides better integration for Please.
+
+### What is an _action_?
+
+As you've seen in the previous examples, workflows consist of _steps_.
+A workflow step can be as simple as a shell script:
+
+```yaml
+- name: Test
+  run: ./pleasew build //...
```

Comment:
Should this be `./pleasew cover //...`? 

I also wonder if we should recommend something like `plz query changes --since master --include_dependencies=transitive`? One of Please's selling points is that it can be used to avoid building/testing stuff that didn't actually get changed.

Reply:
I haven't actually used that before. Maybe it could be part of this code lab, but I was kinda planning to write another one for advanced use cases, like caching.

Reply:
Well I think it should at least be `plz test` or `plz cover`

Reply:
Changed it to test. `plz query changes --since master --include_dependencies=transitive | plz cover -` doesn't work for me: `unknown flag `include_dependencies'`